### PR TITLE
[Compute] `az disk/snapshot`: Bump api-version for disk to support new features

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

close: https://github.com/Azure/azure-cli/issues/27697

**Testing Guide**

**History Notes**

`az disk create`: Add new parameter `--optimized-for-frequent-attach` to improve reliability and performance of data disks that are frequently attached
`az disk/snapshot create`: Add new parameter `--elastic-san-resource-id` to support creating through the ARM id of elastic san volume snapshot

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -207,6 +207,9 @@ def load_arguments(self, _):
         c.argument('support_hibernation', arg_type=get_three_state_flag(), help='Indicate the OS on a disk supports hibernation.', min_api='2020-12-01')
         c.argument('architecture', arg_type=get_enum_type(self.get_models('Architecture', operation_group='disks')), min_api='2021-12-01', help='CPU architecture.')
         c.argument('data_access_auth_mode', arg_type=get_enum_type(['AzureActiveDirectory', 'None']), min_api='2021-12-01', help='Specify the auth mode when exporting or uploading to a disk or snapshot.')
+        c.argument('optimized_for_frequent_attach', arg_type=get_three_state_flag(), min_api='2023-04-02', options_list=['--optimized-for-frequent-attach', '--optimized'],
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:210]: I think this option name `--optimized` is too generic, so I suggest not naming it for a specific feature.
```suggestion
        c.argument('optimized_for_frequent_attach', arg_type=get_three_state_flag(), min_api='2023-04-02', options_list=['--optimized-for-frequent-attach'],
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -218,6 +221,8 @@ def load_arguments(self, _):
         c.argument('upload_type', arg_type=get_enum_type(['Upload', 'UploadWithSecurityData']), min_api='2018-09-30',
                    help="Create the disk for upload scenario. 'Upload' is for Standard disk only upload. 'UploadWithSecurityData' is for OS Disk upload along with VM Guest State. Please note the 'UploadWithSecurityData' is not valid for data disk upload, it only to be used for OS Disk upload at present.")
         c.argument('performance_plus', arg_type=get_three_state_flag(), min_api='2022-07-02', help='Set this flag to true to get a boost on the performance target of the disk deployed. This flag can only be set on disk creation time and cannot be disabled after enabled')
+        c.argument('elastic_san_resource_id', min_api='2023-04-02', options_list=['--elastic-san-resource-id', '--elastic-san-id'],
+                   help='Required if createOption is CopyFromSanSnapshot. This is the ARM id of the source elastic san volume snapshot.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:225]: Could you tell customers how to determine if their `createOption` is `CopyFromSanSnapshot`?

**Comment by zhoxing-ms** [comment]
@ZengTaoxu Due to the inclusion of some new features in this PR, please add the history notes of these new features in the PR title or PR description

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (ZengTaoxu) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:225]: If the argument passed in is --elastic-san-resource-id, createOption is CopyFromSanSnapshot.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:226]: > Required if createOption is CopyFromSanSnapshot.

Isn't this sentence referring to if `createOption` is `CopyFromSanSnapshot`, then the `--elastic-san-resource-id` is required?

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Since the bump version of PR will affect the recording files of PR #27596, merge this PR first to unblock PR #27596.
If the server team discovers some issues from the private package of edge build, then we can submit a new PR to fix them.

**Pull request merged**